In [11]:
import pandas as pd

In [12]:
df_imp = pd.read_csv('Intermediate_partially_cleaned_date_for_stat_testing.csv')
df_imp.head()

Unnamed: 0            StartDate              EndDate          Status  \
0           2  2024-02-18 18:58:07  2024-02-18 19:00:07  Survey Preview   
1           3  2024-02-19 07:02:59  2024-02-19 07:06:19      IP Address   
2           4  2024-02-19 13:39:44  2024-02-19 13:42:55      IP Address   
3           5  2024-02-19 13:49:26  2024-02-19 13:51:57  Survey Preview   
4           6  2024-02-20 08:19:21  2024-02-20 08:21:17      IP Address   

        IPAddress  Progress  Duration (in seconds)  Finished  \
0               0       100                    120      True   
1   73.236.32.238       100                    199      True   
2   74.111.97.243       100                    190      True   
3               0       100                    151      True   
4  71.112.217.136       100                    116      True   

          RecordedDate         ResponseId  ...  Q2041_2  Q2041_3  Q2041_4  \
0  2024-02-18 19:00:09  R_3Pw7bp0cXF8d6Eq  ...        0        0        0   
1  2024-02-19 07:06:20  R_5JmoXVKSvfaPI7T  ...        0        0        0   
2  2024-02-19 13:42:55  R_3nl5gbAd0cY0Gl3  ...        0        0        0   
3  2024-02-19 13:51:58  R_3gMVJhh3CCGGBde  ...        0        0        0   
4  2024-02-20 08:21:18  R_5QK0NLNWSkDunBY  ...        0        0        0   

   Q2041_5  Q2041_6  Q2041_6_TEXT Treatment_Apple_Music Control_Apple_Music  \
0        0        0             0                     0                   1   
1        0        0             0                     0                   1   
2        0        0             0                     0                   1   
3        0        0             0                     1                   0   
4        0        0             0                     1                   0   

  Treatment_Spotify Control_Spotify  
0                 0               0  
1                 0               0  
2                 0               0  
3                 0               0  
4                 0               0  

[5 rows x 81 columns]

## ____________________________________________________________________________

### Statistical Analysis

Added by Niharika. 

Considerations: Imputed Dataframe (df_imp) without any type changes are used for this.

In [13]:
# Statistical Analysis
df = df_imp.copy()

In [14]:
df.columns

Index(['Unnamed: 0', 'StartDate', 'EndDate', 'Status', 'IPAddress', 'Progress',
       'Duration (in seconds)', 'Finished', 'RecordedDate', 'ResponseId',
       'RecipientLastName', 'RecipientFirstName', 'RecipientEmail',
       'ExternalReference', 'LocationLatitude', 'LocationLongitude',
       'DistributionChannel', 'UserLanguage', 'Q1', 'Q3', 'Q3_3_TEXT', 'Q4_1',
       'Q4_2', 'Q4_3', 'Q4_4', 'Q4_5', 'Q4_6', 'Q4_7', 'Q5_1', 'Q5_2', 'Q5_3',
       'Q5_4', 'Q5_5', 'Q6_1', 'Q6_2', 'Q6_3', 'Q6_4', 'Q7', 'Q8', 'Q8_6_TEXT',
       'Q19_1', 'Q19_2', 'Q19_3', 'Q19_4', 'Q19_5', 'Q19_6', 'Q19_6_TEXT',
       'Q10_1', 'Q10_2', 'Q10_3', 'Q10_4', 'Q10_5', 'Q10_6', 'Q10_7', 'Q10_8',
       'Q11', 'Q12', 'Q13', 'Q14', 'Q18', 'Q21', 'Q19_1.1', 'Q19_2.1',
       'Q19_3.1', 'Q19_4.1', 'Q19_5.1', 'Q19_6.1', 'Q19_6_TEXT.1', 'Q2039',
       'Q2040', 'Q2041_1', 'Q2041_2', 'Q2041_3', 'Q2041_4', 'Q2041_5',
       'Q2041_6', 'Q2041_6_TEXT', 'Treatment_Apple_Music',
       'Control_Apple_Music', 'Treatment

In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 47 entries, 0 to 46
Data columns (total 81 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Unnamed: 0             47 non-null     int64  
 1   StartDate              47 non-null     object 
 2   EndDate                47 non-null     object 
 3   Status                 47 non-null     object 
 4   IPAddress              47 non-null     object 
 5   Progress               47 non-null     int64  
 6   Duration (in seconds)  47 non-null     int64  
 7   Finished               47 non-null     bool   
 8   RecordedDate           47 non-null     object 
 9   ResponseId             47 non-null     object 
 10  RecipientLastName      47 non-null     int64  
 11  RecipientFirstName     47 non-null     int64  
 12  RecipientEmail         47 non-null     int64  
 13  ExternalReference      47 non-null     int64  
 14  LocationLatitude       47 non-null     float64
 15  Location

In [16]:
# Success Metrics

# Intent to try out the advertised streaming service
df_ttest = df.copy()

treatment_condition = ((df_ttest['Treatment_Apple_Music'] == 1) | (df_ttest['Treatment_Spotify'] == 1))
control_condition = ((df_ttest['Control_Apple_Music'] == 1) | (df_ttest['Control_Spotify'] == 1))

# Conditions for observing positive intent for 'try'ing advertised streaming service
try_condition = ((df_ttest['Q18'] == 'Maybe') | (df_ttest['Q18'] == 'Yes') | (df_ttest['Q2039'] == 'Maybe') | (df_ttest['Q2039'] == 'Yes'))

# Conditions for observing positive intent for 'switch'ing to the advertised streaming service
switch_condition = ((df_ttest['Q21'] == 'Maybe') | (df_ttest['Q21'] == 'Yes') | (df_ttest['Q2040'] == 'Maybe') | (df_ttest['Q2040'] == 'Yes'))

# Conditions for change in preference order
# treatment
pref_change_condition_treatment = (((df_ttest['Treatment_Apple_Music'] == 1) & (df_ttest['Q19_2'] > df_ttest['Q19_2.1']) ) | 
                                   ((df_ttest['Treatment_Spotify'] == 1) & (df_ttest['Q19_1'] > df_ttest['Q2041_1'])) )

# control
pref_change_condition_control = (((df_ttest['Control_Apple_Music'] == 1) & (df_ttest['Q19_2'] > df_ttest['Q19_2.1']) ) | 
                                   ((df_ttest['Control_Spotify'] == 1) & (df_ttest['Q19_1'] > df_ttest['Q2041_1'])) )

# Applying conditions
df_ttest['positive_intent_try_treatment'] = (treatment_condition & try_condition).astype(int)
df_ttest['positive_intent_try_control'] = (control_condition & try_condition).astype(int)
df_ttest['positive_intent_switch_treatment'] = (treatment_condition & switch_condition).astype(int)
df_ttest['positive_intent_switch_control'] = (control_condition & switch_condition).astype(int)
df_ttest['positive_intent_pref_treatment'] = (pref_change_condition_treatment).astype(int)
df_ttest['positive_intent_pref_control'] = (pref_change_condition_control).astype(int)

df_ttest['positive_intent_treatment'] = (df_ttest['positive_intent_try_treatment'] | df_ttest['positive_intent_switch_treatment'] | df_ttest['positive_intent_pref_treatment']).astype(int)
df_ttest['positive_intent_control'] = (df_ttest['positive_intent_try_control'] | df_ttest['positive_intent_switch_control'] | df_ttest['positive_intent_pref_control']).astype(int)

df_ttest.iloc[:6,-8:]

positive_intent_try_treatment  positive_intent_try_control  \
0                              0                            1   
1                              0                            1   
2                              0                            1   
3                              0                            0   
4                              1                            0   
5                              0                            1   

   positive_intent_switch_treatment  positive_intent_switch_control  \
0                                 0                               0   
1                                 0                               1   
2                                 0                               0   
3                                 0                               0   
4                                 1                               0   
5                                 0                               1   

   positive_intent_pref_treatment  positive_intent_pref_control  \
0                               0                             0   
1                               0                             1   
2                               0                             1   
3                               0                             0   
4                               0                             0   
5                               0                             0   

   positive_intent_treatment  positive_intent_control  
0                          0                        1  
1                          0                        1  
2                          0                        1  
3                          0                        0  
4                          1                        0  
5                          0                        1

In [17]:
from scipy.stats import ttest_ind
import numpy as np

treatment_condition = ((df_ttest['Treatment_Apple_Music'] == 1) | (df_ttest['Treatment_Spotify'] == 1))
control_condition = ((df_ttest['Control_Apple_Music'] == 1) | (df_ttest['Control_Spotify'] == 1))

positive_intent_treatment = df_ttest[treatment_condition]['positive_intent_treatment']
positive_intent_control = df_ttest[control_condition]['positive_intent_control']

print('\n****** Statistical Analysis for Positive Effect on Intent of Survey Takers Advertised Streaming Service ******')

print('\n******Summary Stats for Treatment Group******')
print(positive_intent_treatment.describe())
print('\n******Summary Stats for Control Group******')
print(positive_intent_control.describe())
print('\n******Difference in Averages******')
diff = positive_intent_treatment[:-1].mean() - positive_intent_control[:-1].mean()
print(diff)
print('\n******T-Stat and P-Value Respectively******')
t_stat, p_value = ttest_ind(positive_intent_treatment, positive_intent_control, equal_var=False)
print(t_stat, p_value)


****** Statistical Analysis for Positive Effect on Intent of Survey Takers Advertised Streaming Service ******

******Summary Stats for Treatment Group******
count    20.000000
mean      0.500000
std       0.512989
min       0.000000
25%       0.000000
50%       0.500000
75%       1.000000
max       1.000000
Name: positive_intent_treatment, dtype: float64

******Summary Stats for Control Group******
count    27.000000
mean      0.777778
std       0.423659
min       0.000000
25%       1.000000
50%       1.000000
75%       1.000000
max       1.000000
Name: positive_intent_control, dtype: float64

******Difference in Averages******
-0.2429149797570851

******T-Stat and P-Value Respectively******
-1.973803195262324 0.05605647636172441


In [18]:
from scipy.stats import ttest_ind
import numpy as np

treatment_condition = ((df_ttest['Treatment_Apple_Music'] == 1) | (df_ttest['Treatment_Spotify'] == 1))
control_condition = ((df_ttest['Control_Apple_Music'] == 1) | (df_ttest['Control_Spotify'] == 1))

positive_intent_try_treatment = df_ttest[treatment_condition]['positive_intent_try_treatment']
positive_intent_try_control = df_ttest[control_condition]['positive_intent_try_control']

print('\n****** Statistical Analysis for Intent to TRY Advertised Streaming Service ******')

print('\n******Summary Stats for Treatment Group******')
print(positive_intent_try_treatment.describe())
print('\n******Summary Stats for Control Group******')
print(positive_intent_try_control.describe())
print('\n******Difference in Averages******')
diff = positive_intent_try_treatment[:-1].mean() - positive_intent_try_control[:-1].mean()
print(diff)
print('\n******T-Stat and P-Value Respectively******')
t_stat, p_value = ttest_ind(positive_intent_try_treatment, positive_intent_try_control, equal_var=False)
print(t_stat, p_value)


****** Statistical Analysis for Intent to TRY Advertised Streaming Service ******

******Summary Stats for Treatment Group******
count    20.000000
mean      0.400000
std       0.502625
min       0.000000
25%       0.000000
50%       0.000000
75%       1.000000
max       1.000000
Name: positive_intent_try_treatment, dtype: float64

******Summary Stats for Control Group******
count    27.000000
mean      0.703704
std       0.465322
min       0.000000
25%       0.000000
50%       1.000000
75%       1.000000
max       1.000000
Name: positive_intent_try_control, dtype: float64

******Difference in Averages******
-0.27125506072874495

******T-Stat and P-Value Respectively******
-2.113389785437047 0.040979577495250524


In [19]:
from scipy.stats import ttest_ind
import numpy as np

treatment_condition = ((df_ttest['Treatment_Apple_Music'] == 1) | (df_ttest['Treatment_Spotify'] == 1))
control_condition = ((df_ttest['Control_Apple_Music'] == 1) | (df_ttest['Control_Spotify'] == 1))

positive_intent_switch_treatment = df_ttest[treatment_condition]['positive_intent_switch_treatment']
positive_intent_switch_control = df_ttest[control_condition]['positive_intent_switch_control']

print('\n****** Statistical Analysis for Intent to SWITCH to Advertised Streaming Service ******')

print('\n******Summary Stats for Treatment Group******')
print(positive_intent_switch_treatment.describe())
print('\n******Summary Stats for Control Group******')
print(positive_intent_switch_control.describe())
print('\n******Difference in Averages******')
diff = positive_intent_switch_treatment[:-1].mean() - positive_intent_switch_control[:-1].mean()
print(diff)
print('\n******T-Stat and P-Value Respectively******')
t_stat, p_value = ttest_ind(positive_intent_switch_treatment, positive_intent_switch_control, equal_var=False)
print(t_stat, p_value)


****** Statistical Analysis for Intent to SWITCH to Advertised Streaming Service ******

******Summary Stats for Treatment Group******
count    20.000000
mean      0.250000
std       0.444262
min       0.000000
25%       0.000000
50%       0.000000
75%       0.250000
max       1.000000
Name: positive_intent_switch_treatment, dtype: float64

******Summary Stats for Control Group******
count    27.000000
mean      0.481481
std       0.509175
min       0.000000
25%       0.000000
50%       0.000000
75%       1.000000
max       1.000000
Name: positive_intent_switch_control, dtype: float64

******Difference in Averages******
-0.2368421052631579

******T-Stat and P-Value Respectively******
-1.658923690217617 0.10428980544475337


In [20]:
from scipy.stats import ttest_ind
import numpy as np

treatment_condition = ((df_ttest['Treatment_Apple_Music'] == 1) | (df_ttest['Treatment_Spotify'] == 1))
control_condition = ((df_ttest['Control_Apple_Music'] == 1) | (df_ttest['Control_Spotify'] == 1))

positive_intent_pref_treatment = df_ttest[treatment_condition]['positive_intent_pref_treatment']
positive_intent_pref_control = df_ttest[control_condition]['positive_intent_pref_control']

print('\n****** Statistical Analysis for Intent to CHANGE IN PREFERENCE of the Advertised Streaming Service ******')

print('\n******Summary Stats for Treatment Group******')
print(positive_intent_pref_treatment.describe())
print('\n******Summary Stats for Control Group******')
print(positive_intent_pref_control.describe())
print('\n******Difference in Averages******')
diff = positive_intent_pref_treatment[:-1].mean() - positive_intent_pref_control[:-1].mean()
print(diff)
print('\n******T-Stat and P-Value Respectively******')
t_stat, p_value = ttest_ind(positive_intent_pref_treatment, positive_intent_pref_control, equal_var=False)
print(t_stat, p_value)


****** Statistical Analysis for Intent to CHANGE IN PREFERENCE of the Advertised Streaming Service ******

******Summary Stats for Treatment Group******
count    20.000000
mean      0.150000
std       0.366348
min       0.000000
25%       0.000000
50%       0.000000
75%       0.000000
max       1.000000
Name: positive_intent_pref_treatment, dtype: float64

******Summary Stats for Control Group******
count    27.000000
mean      0.296296
std       0.465322
min       0.000000
25%       0.000000
50%       0.000000
75%       1.000000
max       1.000000
Name: positive_intent_pref_control, dtype: float64

******Difference in Averages******
-0.14979757085020245

******T-Stat and P-Value Respectively******
-1.2054045234017756 0.2343761899406879


##### Results
1. Cumulative Positive Intent: 
    Observed Difference between treatment and control: 4.85% (t-stat = 0.4916, insignificant at 90% and 95% Confidence levels)

2. Positive intent to try the advertised streaming service
    Observed Difference between treatment and control: -27% (t-stat = -2.11, significant at 90% and 95% Confidence levels)

3. Positive intent to Switch to the advertised streaming service
    Observed Difference between treatment and control: -23.68% (t-stat = -1.65, insignificant at 90% and 95% Confidence levels)

4. Positive intent measured by change in preference of the advertised streaming service
    Observed Difference between treatment and control: 4.45% (t-stat = 0.5424, insignificant at 90% and 95% Confidence levels)

##### Inference
- The campaign had a statistically significant negative impact on the intent to try the advertised music streaming service, which is a critical concern. It suggests that something about the campaign or the service as presented may have actively discouraged potential users.
- The lack of significant effects on the intent to switch and the change in preference rankings suggests that the campaign did not effectively persuade existing users of other services to consider switching or altering their preferences in favor of the advertised service.
- The overall effectiveness of the advertisement in encouraging positive perceptions or actions towards the music streaming service is questioned, particularly given the significant negative impact on the willingness to try the service. This indicates a need to reassess the campaign's messaging, targeting, or the perceived value of the service offered to better align with potential users' expectations and preferences.


## ____________________________________________________________________________

Next Steps:
1. Regression Analysis?
2. Sub-strata Analysis?


## ____________________________________________________________________________

#### Questions Identifier

- StartDate - Start Dat
- EndDate - End Date
- Status - Response Type
- IPAddress - IP Address
- Progress - Progress
- Duration (in seconds) - Duration (in seconds)
- Finished - Finished
- RecordedDate - Recorded Date
- ResponseId - Response ID
- RecipientLastName - Recipient Last Name
- RecipientFirstName - Recipient First Name
- RecipientEmail - Recipient Email
- ExternalReference - External Data Reference
- LocationLatitude - Location Latitude
- LocationLongitude - Location Longitude
- DistributionChannel - Distribution Channel
- UserLanguage - User Language
- Q1 - What is your age?
- Q3 - Are you an iPhone or Android user? - Selected Choice
- Q3_3_TEXT - Are you an iPhone or Android user? - Other: - Text
- Q4_1 - Do you own any Apple products? (Check all that apply) - iPhone
- Q4_2 - Do you own any Apple products? (Check all that apply) - iPad
- Q4_3 - Do you own any Apple products? (Check all that apply) - Mac (e.g MacBook, iMac)
- Q4_4 - Do you own any Apple products? (Check all that apply) - Apple Watch
- Q4_5 - Do you own any Apple products? (Check all that apply) - AirPods or other Apple audio devices
- Q4_6 - Do you own any Apple products? (Check all that apply) - Apple TV
- Q4_7 - Do you own any Apple products? (Check all that apply) - None
- Q5_1 - If you own an Apple product, have you enjoyed any free subscription as part of the purchase? (Check all that apply) - iPhone
- Q5_2 - If you own an Apple product, have you enjoyed any free subscription as part of the purchase? (Check all that apply) - MacBook Air/Pro
- Q5_3 - If you own an Apple product, have you enjoyed any free subscription as part of the purchase? (Check all that apply) - iPad X/Air/Pro
- Q5_4 - If you own an Apple product, have you enjoyed any free subscription as part of the purchase? (Check all that apply) - Mac/iMac
- Q5_5 - If you own an Apple product, have you enjoyed any free subscription as part of the purchase? (Check all that apply) - None
- Q6_1 - Have you used any of the free subscriptions offered with your Apple product(s)? (Check all that apply) - Yes, regularly
- Q6_2 - Have you used any of the free subscriptions offered with your Apple product(s)? (Check all that apply) - Yes, occasionally
- Q6_3 - Have you used any of the free subscriptions offered with your Apple product(s)? (Check all that apply) - No, I haven't used any free subscriptions
- Q6_4 - Have you used any of the free subscriptions offered with your Apple product(s)? (Check all that apply) - I don't own an Apple product
- Q7 - How many hours per week do you spend listening to music and podcasts?
- Q8 - Which is your preferred apps for listening to music? - Selected Choice
- Q8_6_TEXT - Which is your preferred apps for listening to music? - Other - Text
- Q19_1 - Rank your preferred apps for listening to music. (Please rank them as Primary, Secondary, Tertiary) - Spotify
- Q19_2 - Rank your preferred apps for listening to music. (Please rank them as Primary, Secondary, Tertiary) - Apple Music
- Q19_3 - Rank your preferred apps for listening to music. (Please rank them as Primary, Secondary, Tertiary) - YouTube Music
- Q19_4 - Rank your preferred apps for listening to music. (Please rank them as Primary, Secondary, Tertiary) - Amazon Music
- Q19_5 - Rank your preferred apps for listening to music. (Please rank them as Primary, Secondary, Tertiary) - Pandora
- Q19_6 - Rank your preferred apps for listening to music. (Please rank them as Primary, Secondary, Tertiary) - Other
- Q19_6_TEXT - Rank your preferred apps for listening to music. (Please rank them as Primary, Secondary, Tertiary) - Other - Text
- Q10_1 - What features do you value most in a music streaming platform? (Select all that apply) - Affordability
- Q10_2 - What features do you value most in a music streaming platform? (Select all that apply) - Audio quality
- Q10_3 - What features do you value most in a music streaming platform? (Select all that apply) - Ease of use
- Q10_4 - What features do you value most in a music streaming platform? (Select all that apply) - Music discovery
- Q10_5 - What features do you value most in a music streaming platform? (Select all that apply) - Personalized playlists
- Q10_6 - What features do you value most in a music streaming platform? (Select all that apply) - Loyalty (years of usage)
- Q10_7 - What features do you value most in a music streaming platform? (Select all that apply) - Unique features
- Q10_8 - What features do you value most in a music streaming platform? (Select all that apply) - Social aspect (my friends use it)
- Q11 - Do you engage in social listening, such as having joint playlists with friends?
- Q12 - What is your monthly budget for entertainment, including music streaming services?
- Q13 - What is the maximum amount you're willing to pay for a music streaming service per month?
- Q14 - How much do you think a subscription to Apple Music costs per month?
- Q18 - Based on the information you just found out do you feel inclined to try out to Apple Music?
- Q21 - Based on the information you just found out do you feel inclined to switch to Apple Music?
- Q19_1.1 - How would you redefine your rank from above what would your new ranking be. Can you rank order your new preference? - Spotify
- Q19_2.1 - How would you redefine your rank from above what would your new ranking be. Can you rank order your new preference? - Apple Music
- Q19_3.1 - How would you redefine your rank from above what would your new ranking be. Can you rank order your new preference? - Amazon Music
- Q19_4.1 - How would you redefine your rank from above what would your new ranking be. Can you rank order your new preference? - YouTube Music
- Q19_5.1 - How would you redefine your rank from above what would your new ranking be. Can you rank order your new preference? - Pandora
- Q19_6.1 - How would you redefine your rank from above what would your new ranking be. Can you rank order your new preference? - Other
- Q19_6_TEXT.1 - How would you redefine your rank from above what would your new ranking be. Can you rank order your new preference? - Other - Text
- Q2039 - Based on the information you just found out do you feel inclined to try out to Spotify?
- Q2040 - Based on the information you just found out do you feel inclined to switch to Spotify?
- Q2041_1 - How would you redefine your rank from above what would your new ranking be. Can you rank order your new preference? - Spotify
- Q2041_2 - How would you redefine your rank from above what would your new ranking be. Can you rank order your new preference? - Apple Music
- Q2041_3 - How would you redefine your rank from above what would your new ranking be. Can you rank order your new preference? - Amazon Music
- Q2041_4 - How would you redefine your rank from above what would your new ranking be. Can you rank order your new preference? - YouTube Music
- Q2041_5 - How would you redefine your rank from above what would your new ranking be. Can you rank order your new preference? - Pandora
- Q2041_6 - How would you redefine your rank from above what would your new ranking be. Can you rank order your new preference? - Other
- Q2041_6_TEXT - How would you redefine your rank from above what would your new ranking be. Can you rank order your new preference? - Other - Text
- Treatment-AppleMusic_DO_QID2031 - Treatment - Apple Music - Display Order - QID2031
- Treatment-AppleMusic_DO_QID2021 - Treatment - Apple Music - Display Order - QID2021
- Treatment/ControlBlock-Spotify_DO_Q2037 - Treatment/Control Block - Spotify - Display Order - Q2037
- Treatment/ControlBlock-Spotify_DO_Q2038 - Treatment/Control Block - Spotify - Display Order - Q2038